cmd/runner/main.go

```diff
@@ -53,16 +70,294 @@ func run(args []string) error {
+	}
+
+	if err := destF.Close(); err != nil {
+		return fmt.Errorf("failed to close dest file: %w", err)
+	}
+
+	return nil
+}
+
+func (h *handler) processPortfolio(ctx context.Context, req *task.ProcessPortfolioRequest) error {
+	// Load the portfolio from blob storage, place it in /mnt/raw_portfolios, where
+	// the `process_portfolios.R` script expects it to be.
+	for _, assetID := range req.AssetIDs {
+		srcURI := blob.Join(h.blob.Scheme(), h.sourcePortfolioContainer, assetID)
+		// TODO: Probably set the CSV extension in the signed upload URL instead.
+		destPath := filepath.Join("/", "mnt", "raw_portfolios", assetID+".csv")
+
+		if err := h.downloadBlob(ctx, srcURI, destPath); err != nil {
+			return fmt.Errorf("failed to download raw portfolio blob: %w", err)
+		}
+	}
+
+	processedDir := filepath.Join("/", "mnt", "processed_portfolios")
+	if err := os.MkdirAll(processedDir, 0600); err != nil {
+		return fmt.Errorf("failed to create directory to download blob to: %w", err)
+	}
+
+	var stdout, stderr bytes.Buffer
+	cmd := exec.CommandContext(ctx, "/usr/local/bin/Rscript", "/app/process_portfolios.R")
+	cmd.Stdout = io.MultiWriter(os.Stdout, &stdout)
+	cmd.Stderr = io.MultiWriter(os.Stderr, &stderr)
+
+	if err := cmd.Run(); err != nil {
+		return fmt.Errorf("failed to run process_portfolios script: %w", err)
+	}
+
+	sc := bufio.NewScanner(&stderr)
 
-	portfolioID := pacta.PortfolioID(os.Getenv("PORTFOLIO_ID"))
-	logger.Info("running PACTA task", zap.String("portfolio_id", string(portfolioID)))
-	if err := h.Execute(ctx, &task.StartRunRequest{
-		PortfolioID: portfolioID,
-	}); err != nil {
-		return fmt.Errorf("failed to run task: %w", err)
+	// TODO: Load from the output database file (or similar, like reading the processed_portfolios dir) instead of parsing stderr
+	var paths []string
+	for sc.Scan() {
+		line := sc.Text()
+		idx := strings.Index(line, "writing to file: " /* 17 chars */)
+		if idx == -1 {
+			continue
+		}
+		paths = append(paths, strings.TrimSpace(line[idx+17:]))
+	}
+
+	// NOTE: This code could benefit from some concurrency, but I'm opting not to prematurely optimize.
+	for _, p := range paths {
+		destURI := blob.Join(h.blob.Scheme(), h.destPortfolioContainer, filepath.Base(p))
+		if err := h.uploadBlob(ctx, p, destURI); err != nil {
+			return fmt.Errorf("failed to copy processed portfolio from %q to %q: %w", p, destURI, err)
+		}
 	}
 
 	return nil
 }
+
+func createReportReq() (*task.CreateReportRequest, error) {
+	pID := os.Getenv("PORTFOLIO_ID")
+	if pID == "" {
+		return nil, errors.New("no PORTFOLIO_ID was given")
+	}
+
+	return &task.CreateReportRequest{
+		PortfolioID: pacta.PortfolioID(pID),
+	}, nil
+}
+
+func (h *handler) createReport(ctx context.Context, req *task.CreateReportRequest) error {
+	baseName := string(req.PortfolioID) + ".json"
+
+	// Load the processed portfolio from blob storage, place it in /mnt/
+	// processed_portfolios, where the `create_report.R` script expects it
+	// to be.
```

Comment:
I'm probably just missing context that you have - is a portfolio report always generated off of exactly one processed portfolio? I thought it (like above) could be multiple?

Reply:
Great question! I actually don't know, and I expect this code will require a bunch of tweaks once [Alex's stub interface](https://github.com/AlexAxthelm/stub_PACTA_containers_for_app/) gets reconciled with [the actual code interface](https://github.com/RMI-PACTA/workflow.transition.monitor)